File: src/org/jgll/recognizer/PrefixGLLRecognizer.java
package org.jgll.recognizer;

import org.jgll.grammar.GrammarSlot;

public class PrefixGLLRecognizer extends AbstractGLLRecognizer {
	
	@Override
	public void add(GrammarSlot slot, GSSNode u, int inputIndex) {
		if(slot == startSlot && u == u0) {
			descriptorStack.clear();
			recognized = true;
			return;
		}
		super.add(slot, u, inputIndex);
	}

}

File: src-scala/org/jgll/recognizer/PrefixGLLRecognizer.scala
package org.jgll.recognizer

import org.jgll.grammar.slot.GrammarSlotTrait

trait PrefixGLLRecognizerTrait {
  self: AbstractGLLRecognizerTrait
   with GrammarSlotTrait
   with GSSNodeTrait =>
  class PrefixGLLRecognizer extends AbstractGLLRecognizer {

    override def add(slot: GrammarSlot, u: GSSNode, inputIndex: Int) {
      if (slot == AbstractGLLRecognizer.startSlot && u == AbstractGLLRecognizer.u0) {
        descriptorStack.clear()
        recognized = true
        return
      }
      super.add(slot, u, inputIndex)
    }
  }
}